### PR TITLE
Chore: Remove Nav + story IconName type assertions

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

Removes some `icon as IconName` type assertions in favor of the `toIconName` function. 

This PR changes:
 - some stories
 - navigation/page layouts

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/core/components/NavBar/NavBarMenu.tsx [hunk 1]
```diff
@@ -459,3 +459,6 @@ const getCollapsibleStyles = (theme: GrafanaTheme2) => ({
 function linkHasChildren(link: NavModelItem): link is NavModelItem & { children: NavModelItem[] } {
   return Boolean(link.children && link.children.length > 0);
 }
+function toIconName(icon: string | undefined): IconName | undefined {
+  throw new Error('Function not implemented.');
+}
```
Comment (ashharrison90) [inline public/app/core/components/NavBar/NavBarMenu.tsx:464]: i don't understand this bit?

**Review by ashharrison90 — COMMENTED** [review]
one question, otherwise lgtm

File: public/app/core/components/NavBar/NavBarMenu.tsx (hunk 1 above)
Comment (joshhunt) [inline public/app/core/components/NavBar/NavBarMenu.tsx:464]: we don't talk about this

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌

File: public/app/core/components/NavBar/NavBarMenu.tsx (hunk 1 above)
Comment (joshhunt) [reply public/app/core/components/NavBar/NavBarMenu.tsx]: I pressed the wrong button. its fixed now 😌 

image

**Pull request merged**